docs/data/date-pickers/adapters-locale/UTCLuxon.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { DateTime } from 'luxon';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+
+export default function UTCLuxon() {
+  const [value, setValue] = React.useState<DateTime | null>(null);
+
+  return (
+    <LocalizationProvider
+      dateAdapter={AdapterLuxon}
+      dateLibInstance={{ zone: 'UTC' }}
```

Comment:
This is not super sexy
I had a cleaner idea for luxon, which is to add static methods on the adapter which create an instance of the adapter with some custom properties.

Something like:

```tsx
// Set UTC
<LocalizationProvider dateAdapter={AdapterLuxon.withZone('UTC')}>

// Set a custom timezone
<LocalizationProvider dateAdapter={AdapterLuxon.withZone('Pacific/Auckland')}>
```

I allow us to get rid of the `dateLibInstance` which is not typeable.

I'm asking this because the more I work with timezones, the uglier those instances gets (take a look at [the codesandbox in this comment](https://github.com/mui/mui-x/issues/7832#issuecomment-1469910841)).

Migrating smoothly to something like `AdapterDayjs.withZone('Pacific/Auckland')` would be cleaner imho.
It would abstract those instances into a shared approach we can more easily document.


But I would have to check that it is possible to end up with the same API for every adapter.
So maybe it's good to start with a `dateLibInstance` approach, release the new approach during v6 and drop the old one in v7.